neuraxle/data_container.py

```diff
@@ -113,12 +113,19 @@ def convolved_1d(self, stride, kernel_size) -> Iterable['DataContainer']:
         .. seealso::
             `<https://github.com/guillaume-chevalier/python-conv-lib>`_
         """
-        conv_current_ids = convolved_1d(stride=stride, iterable=self.current_ids, kernel_size=kernel_size)
-        conv_data_inputs = convolved_1d(stride=stride, iterable=self.data_inputs, kernel_size=kernel_size)
-        conv_expected_outputs = convolved_1d(stride=stride, iterable=self.expected_outputs, kernel_size=kernel_size)
+        conv_current_ids = convolved_1d(stride=stride, iterable=self.current_ids, kernel_size=kernel_size, include_incomplete_pass=True)
+        conv_data_inputs = convolved_1d(stride=stride, iterable=self.data_inputs, kernel_size=kernel_size, include_incomplete_pass=True)
+        conv_expected_outputs = convolved_1d(stride=stride, iterable=self.expected_outputs, kernel_size=kernel_size, include_incomplete_pass=True)
```

Comment:
If you updated conv, please open PR in conv and update the requirements txt *and* setup py of neuraxle.

Reply:
@alexbrillant This code will speak more to you than words. The first test has `incomplete_pass=True`, while the second has `incomplete_pass=False`. 

1. The first test, with a window size of 7, will loop on the 10 entries like [range(7), range(7,10)]. 
2. The second test with a window size of 7 and the same 10 entries will loop like [range(7)] and **will miss the last 3 entries**. 

Missing the last 3 entries (as in "2.") was the default behavior in conv 0.1 when the kernel size was bigger than the entries it was looping on. The new flag, if set to `True`, fixes this. 

```

    def test_big_kernel_loops_twice(self):
        expected = tuple(range(7 + 3))
        result = []

        for kernel_hover in convolved_1d(expected, kernel_size=7, padding=0, stride=7, default_value=None,
                                         include_incomplete_pass=True):
            result.append(kernel_hover)

        self.assertEqual(len(result), 2)

    def test_big_kernel_loops_once(self):
        expected = tuple(range(7 + 3))
        result = []

        for kernel_hover in convolved_1d(expected, kernel_size=7, padding=0, stride=7, default_value=None,
                                         include_incomplete_pass=False):
            result.append(kernel_hover)

        self.assertEqual(len(result), 1)

```

Reply:
conv has a bug. it also adds nones at the end of incomplete batches...

Reply:
Just realized this wasn't tested. Will fix in conv later, resolved to issue #186.